### DISCUSSION TITLE
3.0.0

### DISCUSSION THREAD

**Discussion by T8RIN** [T8RIN/ImageToolbox#1333] [Announcements]
## What's Changed
* Make camera unnecessary to run the app
* Added Jpeg 2000 format: .jp2 .j2k
* Added Tiff compression and previewing
* Added 39 new filters:
  - Low Poly
  - Sand Painting
  - Palette Transfer (#1205)
  - Enhanced Oil
  - Simple Old Tv
  - HDR
  - Simple Sketch 
  - Gotham
  - Color Poster
  - Tri Tone
  - Clahe Oklch
  - Clahe Jzazbz
  - Clahe Oklab
  - Yililoma Dithering
  - Clustered 2x2 Dithering
  - Clustered 4x4 Dithering
  - Clustered 8x8 Dithering
  - Polka Dot
  - LUT 512*512
  - Amatorka
  - Miss Etikate
  - Soft Elegance
  - Soft Elegance Variant
  - Cube LUT
  - Palette Transfer Variant
  - Bleach Bypass
  - Candlelight
  - Drop Blues
  - Edgy Amber
  - Fall Colors
  - Film Stock 50
  - Foggy Night
  - Kodak
  - Pop Art
  - Сelluloid 
  - Coffee 
  - Golden Forest
  - Greenish
  - Retro Yellow
* Add Image Splitter option by #724
* Added quality controlling for watermarking option by #1192
* Added Aspect Ratio preset by #1201
* Added choosing between crop/fit in Aspect ratio preset by #1149
* Added ability to Import/Export OCR models by #1196
* Added ability to choose position when using crop resize mode by #1138
* Improved Image Preview option, new grid layout, also added ability to select and process multiple images at once by #1209
* Fix many bugs in crop resize mode preview
* Added ability to move to the next/previous image by swiping in batch modes by #1206
* Added ability to edit OCR extracted text by #1206
* Added ability to reorder favorite filters by #1206
* Added ability to add rotation to rectangle in drawing by #1058
* Fix crashes by #1214
* Speed up dithering filters and some others too
* 8 New scale color spaces
  - Oklab SRGB
  - Oklab Rec.709
  - Oklab Gamma 2.2
  - Oklab Gamma 2.8
  - Jzazbz sRGB
  - Jzazbz Rec.709
  - Jzazbz Gamma 2.2
  - Jzazbz Gamma 2.8
* Added Favorite Options when grouping disabled by #1238
* Fix cipher double click by #1236
* Added ability to capture picture from camera in OCR without changing image picker mode
* Move draw color selection nearer to color picker
* Fix apng crash
* Added Color Tools option
  - Mixing
  - Conversion
  - Harmonies
  - Shading
* Neutral LUT exporting to create own Lookup Tables
* Fixed Camera Permission in Document Scanner by @egmsia01 in #1270
* Fix Preset selector conflicts with quality options by #1257
* Added QOI Image Format support by #1260
* Map JXL effort from 1..10 instead of 0..9 by #1274
* Added ability to enable link previews in OCR and QR Code Scanner
* Added zoom to QR Scanner
* Added scanning from image by #1282
* Fix Watermark Text sizing doesn't match preview by #1276
* Improve quick tiles
* Added ICO image conversion support
* Added GIF to WEBP tool
* Added WEBP tools by #1297
  - WEBP to images
  - Images to WEBP
* Fix JXL Tools layout
* Improve embedded media picker ui
* Added PSD decoding
* Added JXL and QOI support for embedded picker by #1270
* Improve UI of image pagers
* Added Choosing the defaults for the Draw feature (color, mode) by #1259
* Added DJVU images loading
* Added Option to remove timestamp from filename by #1309
* Added Option to use non formatted timestamp
* Imrpove Emoji Selection UI
* Added One time save locations editing to settings also
* Added Recently used colors in selection of them
* Added 10 New Icon Shapes
  - Shuriken
  - Explosion
  - Bookmark
  - Pill
  - Burger
  - Shield
  - Droplet
  - Arrow
  - Egg
  - Map
* Added ability to change QR code shape by #1233
* Added ability to save OCR recognized text into file by #1236
* Improve accessibility of color selection by #1246
* Added Fit Resize mode by #1327
* Added default resize type setting by #744
* Added Ability to control system bars visibility by #1327
* Add Ultra HDR preview support in embedded picker and Image Preview tool by #1124
* Added Noise generator by #948

## New Contributors
* @egmsia01 in #1270
* @oupson in #1306

**Full Changelog**: https://github.com/T8RIN/ImageToolbox/compare/2.9.0...3.0.0

<hr /><em>This discussion was created from the release <a href='https://github.com/T8RIN/ImageToolbox/releases/tag/3.0.0'>3.0.0</a>.</em>

**Comment by NikunjK8** [comment]
Still not updated on play store?

**Reply by T8RIN** [reply]
The play store support is annoying me, they don't allow publishing because i have "manage external storage" permission to view jxl/qoi images, they just ignore the fact that it's fully optional

**Reply by NikunjK8** [reply]
Yeah i remember a lot of devs complaining about this on Reddit. This'll be a pain for sure.

Btw usually i can update apps downloaded from play store with izzy repo and vice versa. But ImageToolbox is signed with different keys, any reason for that?

**Reply by T8RIN** [reply]
F-droid versions compiled by f-droid from source code

**Reply by T8RIN** [reply]
I think that it's better to just remove permission for the play store rather than arguing with bot🐸

**Reply by NikunjK8** [reply]
No, i was asking about izzy repo. As far as i know izzy takes apps directly from github. Fdroid indeed compile them.

**Reply by NikunjK8** [reply]
And you're right. It's better to keep a separate version on play store then working under a risk of removal
